Makefile

```diff
@@ -1,9 +1,12 @@
-
 all: build/spec.pdf
 
 build/spec.pdf: spec.md include/spec-template.tex include/firrtl.xml include/ebnf.xml | build/
 	pandoc $< --template include/spec-template.tex --syntax-definition include/firrtl.xml --syntax-definition include/ebnf.xml -r markdown+table_captions+inline_code_attributes+gfm_auto_identifiers --filter pandoc-crossref -o $@
 
+IMG_SRC=$(wildcard include/img_src/*.dot) 
+images: $(IMG_SRC) | build/ build/img/
+	$(foreach dotfile,$(IMG_SRC),dot -Tpng $(dotfile) -o build/img/$(patsubst %.dot,%.png,$(lastword $(subst /, ,$(dotfile))));)
+
```

Comment:
@seldridge this is the best I could do ;_;

I am not v good with makefiles, please lmk if there is a nicer way

Reply:
You can change the `%/` rule to be `mkdir -p` and simplify the order-only prerequisite to just be `build/img/`.

A slightly more clean way to do this is to use a simpler rule that builds one PNG for one DOT file. I can never remember the wildcard syntax, but the actual files can be made explicit for now. Something like:

```make
IMG_PNG=include/img_src/foo.dot include/img_src/bar.dot
build/img/%.png: include/img_src/%.dot | build/img/
    dot -Tpng $< -o $@
```

You can then add a normal (not order-only) prerequisites to `build/spec.pdf` (and move `IMG_PNG` up to the top of the file:
```make
build/sped.pdf: spec.md include/spec-template.tex include/firrtl.xml include/ebnf.xml $(IMG_PNG)
```